src/Microsoft.Azure.SignalR.AspNet/Infrastructure/ServiceMessageBus.cs

```diff
@@ -2,24 +2,190 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.AspNet.SignalR;
+using Microsoft.AspNet.SignalR.Infrastructure;
+using Microsoft.AspNet.SignalR.Json;
 using Microsoft.AspNet.SignalR.Messaging;
+using Microsoft.AspNet.SignalR.Transports;
+using Microsoft.Azure.SignalR.Protocol;
+using Newtonsoft.Json;
 
 namespace Microsoft.Azure.SignalR.AspNet
 {
-    /// <summary>
-    /// TODO: This class is responsible for transform SignalR Messages to ServiceMessages and send them to the service runtime
-    /// </summary>
     internal class ServiceMessageBus : MessageBus
     {
+        private readonly JsonSerializer _serializer;
+        private readonly IServiceConnectionManager _serviceConnectionManager;
+        private readonly IAckHandler _ackHandler;
+        private readonly IServiceProtocol _serviceProtocol;
+        private readonly IMemoryPool _pool;
+
         public ServiceMessageBus(IDependencyResolver resolver) : base(resolver)
         {
+            _serviceConnectionManager = resolver.Resolve<IServiceConnectionManager>() ?? throw new ArgumentNullException(nameof(IServiceConnectionManager));
+            _serializer = resolver.Resolve<JsonSerializer>() ?? throw new ArgumentNullException(nameof(JsonSerializer));
+            _ackHandler = resolver.Resolve<IAckHandler>() ?? throw new ArgumentNullException(nameof(IAckHandler));
+            _serviceProtocol = resolver.Resolve<IServiceProtocol>() ?? throw new ArgumentNullException(nameof(IServiceProtocol));
+            _pool = resolver.Resolve<IMemoryPool>() ?? throw new ArgumentNullException(nameof(IMemoryPool));
         }
 
         public override Task Publish(Message message)
```

Comment:
Lets move all of the parsing to a separate component that returns "addresses".

Reply:
Move inside a [SignalRMessageParser](https://github.com/Azure/azure-signalr/pull/152/files#diff-88b66ec494a1d6fe9bb4b5c7d496f26f)